docs/design/design_vuln_scan_job_in_same_namespace_of_workload.md

```diff
@@ -0,0 +1,81 @@
+# Run Vulnerability Scan Job in Same namespace of workload
+
+## Overview 
+
+Currently, when image is pulled from cloud registries(ecr, acr) in cluster and authentication is set on service account. Then 
+Trivy plugin cannot use that authentication to pull cloud registry image. If user want to get those images scanned then they
+have to specify those credentials at starboard service account([ECR registry configuration]). Also, recently there was one
+option added in Trivy plugin to use Trivy filesystem scan, in which starboard doesnt have option to reuse ImagePullSecret 
+defined on workload spec or on service account. To resolve this limitation, if starboard manages to run scan job in same 
+namespace where workload is running then we can reuse the service account and imagePullSecret available with the workload.
+
+## Solution
+
+Add an environment variable to operator eg `OPERATOR_ENABLE_SCAN_JOB_IN_SAME_NAMESPACE=true`. This way operator can schedule
```

Comment:
Looks Great !!
Couple of comments: 
- The parameter name is generic (for all jobs), however it applies for `vulnerability scanning` job only, for now, just to make sure this reflected in documentation
- looking forward in the future if we want this env. Var to support other jobs types (kube-bench / confaudit and etc) --> since this job will inherit the service account of the workload, it might have some limitation for other jobs which are not `Scan vulnerability` as different jobs might require different permission

Reply:
thanks Chen, yes agree with you that this option will have some limitations for other scans considering the permission barrier. I will change it to `OPERATOR_ENABLE_VULNERABILITY_SCAN_JOB_IN_SAME_NAMESPACE`.

Reply:
Notice that params related to vulnerability scanners are prefixed with `OPERATOR_VULNERABILITY_SCANNER_`

OPERATOR_VULNERABILITY_SCANNER_ENABLED
OPERATOR_VULNERABILITY_SCANNER_SCAN_ONLY_CURRENT_REVISIONS
OPERATOR_VULNERABILITY_SCANNER_REPORT_TTL